R/generateIIASASubmission.R

```diff
@@ -62,90 +54,75 @@ generateIIASASubmission <- function(mifs = ".", mapping = NULL, model = "REMIND
                               iiasatemplate = iiasatemplate))
   }
 
-  message("\n### Generating .mif and .xlsx files using mapping ", mappingFile, ".")
+  m <- read.csv2(mappingFile) %>%
+    mutate(
+      !!sym("factor") := as.numeric(!!sym("factor")),
+      # this is not optimal and error-prone: we must dissect variable into variable and unit again
+      # could be avoided, if we expect mappings with variable and unit fields
+      # instead of having the unit as part of the variable name
+      !!sym("piam_unit") := gsub("^\\(|\\)$", "", regmatches(!!sym("piam_variable"), regexpr("\\(([^)]*)\\)[^(]*$", !!sym("piam_variable")))), # nolint
+      !!sym("piam_variable") := gsub(" \\(([^)]*)\\)[^(]*$", "", !!sym("piam_variable")), # nolint
+      !!sym("Unit") := gsub("^\\(|\\)$", "", regmatches(!!sym("Variable"), regexpr("\\(([^)]*)\\)[^(]*$", !!sym("Variable")))), # nolint
+      !!sym("Variable") := gsub(" \\(([^)]*)\\)[^(]*$", "", !!sym("Variable")) # nolint
```

Comment:
Just an idea, I think this is more reliable, because Mika put a hell lot of effort in getting that right :)
```suggestion
      !!sym("piam_unit") := magclass::unitsplit(!!sym("piam_variable"))$unit
      !!sym("piam_variable") := magclass::unitsplit(!!sym("piam_variable"))$variable
      !!sym("Unit") := magclass::unitsplit(!!sym("Variable"))$unit
      !!sym("Variable") := magclass::unitsplit(!!sym("Variable"))$variable
```
I agree with you that it is not very clever to have the variables and units put together in the mapping file in the first place… It is done [here in generateMappingfile()](https://github.com/pik-piam/piamInterfaces/blob/master/R/generateMappingfile.R#L113-L114) and we should maybe not do that :)

Why do you choose `Unit` and `Variable` with big letters, when you have to rename it just some lines later to small letters again?

Reply:
Thanks for drawing my attention to `unitsplit`.  

> Why do you choose Unit and Variable with big letters, when you have to rename it just some lines later to small letters again?

If I choose lowercase, i then also should to remove the uppercase variables explicitly, so it doesn't make much of a difference.